### ISSUE TITLE
os.SEEK_{SET,CUR,END} constants are deprecated.

### ISSUE THREAD

**Issue by shurcooL** [shurcooL/vfsgen#21]
See https://godoc.org/os#SEEK_SET:

> Deprecated: Use io.SeekStart, io.SeekCurrent, and io.SeekEnd.

Thanks @dominikh for spotting and reporting.

**Comment by justinsb** [comment]
This commit prevents vfsgen from being used with versions of go prior to 1.7 (because the constants were only introduced in 1.7).

**Comment by shurcooL** [comment]
That is true. Is it causing a problem for you @justinsb? Or is that just an observation?

**Comment by justinsb** [comment]
It's not a huge problem, in that I'm able to stick with go-bindata for the moment (despite its shortcomings), and can easily just pin to a version before this commit when I want to try again, or by then hopefully I'll be on go 1.7.

But go 1.7.0 has a bug that prevents my using it (because I do a lot of work with kubernetes): https://github.com/golang/go/issues/16739  And I think bigger projects are fairly conservative in adopting newer versions of go anyway (though again I'm primarily thinking of kubernetes), so if the only motivation is to avoid a deprecation warning, it seems that it might be better to delay adopting 1.7-specific features in vfsgen.  But I'm sure you're aware of the tradeoffs here :-)

**Comment by shurcooL** [comment]
Got it, that makes sense.

Let me explain what happened. I fixed the bug by pushing 9410553432a2ba3643c754fa50b4fcd0cce966ef and only then realized that those new replacement symbols were only added to `io` in 1.7, not earlier as I imagined.

At that point, I had the option to revert, or stay with it. My personal SLA for my free open source Go code is to always support the current stable release of Go (and GopherJS). So, right now that's 1.7. Normally, I don't go out of my way to stop things from working on older versions of Go, but given my limited resources, I also can't afford to actively support old versions. I can only support people who use latest stable version of Go.

So I decided not to revert because of that, since 1.7.1 is the current stable version of Go.

I can definitely make a branch called `go1.6` that will have the latest version of `vfsgen` that still works on older versions of Go, but I won't be able to actively support or develop it (per my SLA).

Hope that makes sense. :) The above is a rough general rule, I can still make exceptions if there are specific valid needs, etc. So feel free to let me know if you want that `go1.6` branch or if you have other thoughts/requests!
